datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -152,3 +176,34 @@ private fun Questionnaire.Item.createQuestionnaireResponseItem():
         }
     }
 }
+
+/**
+ * Traverse (DFS) through the list of questionnaire items and the list of questionnaire response
+ * items and check if the linkid of the matching pairs of questionnaire item and questionnaire
+ * response item are equal.
+ * The traverse is carried out in the two lists in tandem. The two lists should be structurally
+ * identical.
+ */
+fun validateQuestionnaireResponse(
+    questionnaireItemList: List<Questionnaire.Item>,
+    questionnaireResponseItemList: List<QuestionnaireResponse.Item>
+): Boolean {
+    val questionnaireItemListIterator = questionnaireItemList.iterator()
+    val questionnaireResponseItemListIterator = questionnaireResponseItemList.iterator()
+    while (
+        questionnaireItemListIterator.hasNext() &&
+        questionnaireResponseItemListIterator.hasNext()
+    ) {
+        val questionnaireItem = questionnaireItemListIterator.next()
+        val questionnaireResponseItem = questionnaireResponseItemListIterator.next()
+        if (questionnaireItemListIterator.hasNext()
+                .xor(questionnaireResponseItemListIterator.hasNext()))
```

Comment:
you could also use the kotlin infix notation:

https://kotlinlang.org/docs/functions.html#infix-notation

```suggestion
    questionnaireItemListIterator.hasNext() xor questionnaireResponseItemListIterator.hasNext()
```

Reply:
done